src/daemon/daemon.go

```diff
@@ -280,33 +282,36 @@ type MessageEvent struct {
 // over the quit channel provided to Init.  The Daemon run loop must be stopped
 // before calling this function.
 func (dm *Daemon) Shutdown() {
+	// close the daemon loop first
+	q := make(chan struct{}, 1)
```

Comment:
The `Daemon` loop is listening the `quitC` channel, once received `q` from `quitC`,  will send `struct{}{}` to the `q` and stop the loop. Here we use buffered channel, so that the loop can return immediately, instead of waiting for the `q` channel receiver.